### PR TITLE
ceph.in: drop the compatiiblity to handle non json commands

### PR DESCRIPTION
**Pull request by tchaikov**
we has been using json command since giant. no need to bring this cruft
into luminous.

Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
@Liuchang0812  it's a follow-up of your cleanup work.

**Comment by Liuchang0812** [comment]
:smile:   I'm happy to review and test it.

**Comment by tchaikov** [comment]
that'd be great! thanks.

**Comment by Liuchang0812** [comment]
Hi, Kefu. here are some problems with `ctest -R "cephtool-test-*"   --output-on-failure`:

```
/home/liuchang/WorkSpace/ceph/qa/workunits/cephtool/test.sh:47: expect_false:  ceph mds set allow_new_snaps taco
2017-06-06 19:24:28.577335 7f85cb453700 -1 WARNING: all dangerous and experimental features are enabled.
2017-06-06 19:24:28.593196 7f85cb453700 -1 WARNING: all dangerous and experimental features are enabled.
Traceback (most recent call last):
  File "/home/liuchang/WorkSpace/ceph/build/bin/ceph", line 1102, in <module>
    retval = main()
  File "/home/liuchang/WorkSpace/ceph/build/bin/ceph", line 1037, in main
    verbose)
  File "/home/liuchang/WorkSpace/ceph/build/bin/ceph", line 567, in new_style_command
    inbuf=inbuf)
  File "/home/liuchang/WorkSpace/ceph/src/pybind/ceph_argparse.py", line 1352, in json_command
    argdict, outs))
RuntimeError: "{u'var': u'allow_new_snaps', 'prefix': u'mds set', u'val': u'taco'}": bad command or argument: value must be false|no|0 or true|yes|1
/home/liuchang/WorkSpace/ceph/qa/workunits/cephtool/test.sh:47: expect_false:  return 0
```
It seems that the logical of `return code` is broken.

File: src/pybind/ceph_argparse.py [hunk 1]
```diff
@@ -1347,4 +1347,7 @@ def json_command(cluster, target=('mon', ''), prefix=None, argdict=None,
         else:
             raise
 
+    if ret == -errno.EINVAL:
+        raise RuntimeError('"{0}": bad command or argument: {1}'.format(
```
Comment (Liuchang0812) [inline src/pybind/ceph_argparse.py]: Here is the problem. `return json_command("foo")` will get 0 when json_command raises exception here!

Maybe we shouldn't panic here, how about putting error message to `outs` only.

Comment (tchaikov) [inline src/pybind/ceph_argparse.py]: yeah, i am dropping this hunk.

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-06-08_20:27:41-rados-wip-sage-testing2-distro-basic-smithi/

**Pull request merged**